fastval.go

```diff
@@ -302,9 +302,26 @@ func (val FastVal) ToJsonString() (FastVal, error) {
 	return val, errors.New("invalid type coercion")
 }
 
+func (val FastVal) floatToIntOverflows() bool {
+	floatVal := val.GetFloat()
+
+	// Instead of using math constants that could potentially lead to rounding errors,
+	// force a float-to-float comparison here
+	if !(floatVal >= -9223372036854775808.0 && floatVal < 9223372036854775808.0) {
```

Comment:
better to use MaxInt64 and MinInt64

Reply:
I suppose the rounding error probability is miniscule.

Reply:
you could do float(MaxInt64) before you compare.

Reply:
It's probably not needed

Reply:
how do we verify "92233.." is the right number to use, then? will the test data be based on MaxInt64?

Reply:
I changed the latest changeset to MaxInt64 and MinInt64